python/tskit/trees.py

```diff
@@ -3423,7 +3423,7 @@ def __getstate__(self):
         return self.dump_tables()
 
     def __setstate__(self, tc):
-        self.__init__(tc.tree_sequence().ll_tree_sequence)
+        TreeSequence.__init__(self, tc.tree_sequence().ll_tree_sequence)
```

Comment:
Hmm, so this explicitly skips the subclass init. What are subclasses supposed to do then, I guess define their own ``__setstate__``

Is there any official guidance on what we're suppose to do here?

Reply:
Yes, the proper fix is for pyslim to override `__setstate__` in it's subclass, calling `super().__init__` before doing it's own work.